proxy/src/proxy.rs

```diff
@@ -144,8 +148,19 @@ impl ProxyConnection {
 
                     return Ok((get_param("user")?, get_param("database")?));
                 }
-                // TODO: implement proper stmt cancellation
-                StartupRequestCode::Cancel => bail!("query cancellation is not supported"),
+                StartupRequestCode::Cancel => {
+                    bail!("Cancel message has no key data, ignoring.")
+                    // TODO After connecting using tokio_postgres, send that cancel
+                    //      token to the client. It must happen after AuthenticationOk
+                    //      but before ReadyForQuery.
+                    //      See https://www.postgresql.org/docs/7.2/protocol-protocol.html
```

Comment:
> docs/7.2/

beware of old versions! :) 7.2 is 16 years old postgres. Google rates old docs same as the new ones -- switch to the 'current version' on the docs page.